pkg/gadgets/profile/block-io/tracer/bpf/biolatency.bpf.c

```diff
@@ -64,21 +62,39 @@ static __always_inline int trace_rq_start(struct request *rq, int issue)
 	return 0;
 }
 
+// https://github.com/torvalds/linux/commit/a54895fa057c67700270777f7661d8d3c7fda88a
+// -       TP_PROTO(struct request_queue *q, struct request *rq),
+// +       TP_PROTO(struct request *rq),
+struct request___empty {};
+struct request_queue___empty {};
```

Comment:
Updated

Reply:
I am not sure what you mean, can you clarify?

`struct request` and `struct request_queue` are not defined in cilium/ebpf but in the kernel.

The relocation of type `BPF_TYPE_MATCHES` (with `bpf_core_type_matches`) are implemented in the Linux kernel, but cilium/ebpf chooses to not make use of this and instead reimplements it itself, so it is compatible with older kernel without this type of relocation support. So cilium/ebpf is not free to choose the semantics (whether to check types recursively) but replicate the semantics of the kernel implementation.

The fact that cilium/ebpf initially didn't check the type recursively was a bug because it was not replicating the kernel semantics. But that bug never made it to the main branch because it was fixed during reviews before the PR was merged.

So I don't think we should refer to a behaviour in cilium/ebpf code that was never merged.